editor/modes/visual-editor/block.js

```diff
@@ -11,20 +11,62 @@ import Toolbar from 'components/toolbar';
 import BlockMover from 'components/block-mover';
 import BlockSwitcher from 'components/block-switcher';
 
+const formattingControls = [
+	{
+		icon: 'editor-bold',
+		title: wp.i18n.__( 'Bold' ),
+		format: 'bold'
+	},
+	{
+		icon: 'editor-italic',
+		title: wp.i18n.__( 'Italic' ),
+		format: 'italic'
+	},
+	{
+		icon: 'editor-strikethrough',
+		title: wp.i18n.__( 'Strikethrough' ),
+		format: 'strikethrough'
+	}
+];
+
 class VisualEditorBlock extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
 		this.bindBlockNode = this.bindBlockNode.bind( this );
 		this.setAttributes = this.setAttributes.bind( this );
 		this.maybeDeselect = this.maybeDeselect.bind( this );
 		this.maybeHover = this.maybeHover.bind( this );
+		this.onFormatChange = this.onFormatChange.bind( this );
+		this.toggleFormat = this.toggleFormat.bind( this );
 		this.previousOffset = null;
+		this.state = {
+			formats: {}
+		};
 	}
 
 	bindBlockNode( node ) {
 		this.node = node;
 	}
 
+	onFormatChange( formats ) {
+		if ( ! this.state.hasEditable ) {
```

Comment:
An alternative would be an extra setting for it, but not sure if that's any better.